src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```diff
@@ -160,6 +166,38 @@ object DepositHandler {
     }
   }
 
+  private def resolveFetchItems(bagitDir: File)(implicit id: String): Try[Unit] = Try {
+    log.debug(s"[$id] Checking fetch.txt")
+    getFetchTxt(bagitDir).foreach(fetchText =>
+      { log.debug(s"[$id] Resolving files in fetch.txt")
+        fetchText.asScala.foreach(item =>
+        getUrl(item.getUrl).foreach(url =>
+          Using.urlInputStream(url).foreach(src =>
+            Files.copy(src, Paths.get(bagitDir.getAbsolutePath, item.getFilename)))))
+      }
+    )
+  }
+
+  private def getFetchTxt(bagitDir: File)(implicit id: String): Option[FetchTxt] = Option {
+    getBagFromDir(bagitDir).getFetchTxt
+  }
+
+  private def getBagFromDir(dir: File)(implicit id: String): Bag = {
```

Comment:
I think that we have to call _bagFactory.createBag_ twice, because the contents of the bag directory has been changed after fetching the files in the fetch.txt.